x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/execution/sequence/SequenceMatcher.java

```diff
@@ -72,11 +73,16 @@ public void clear() {
     // Set of completed sequences - separate to avoid polluting the other stages
     // It is a set since matches are ordered at insertion time based on the ordinal of the first entry
     private final Set<Sequence> completed;
+    private final Set<Sequence> toCheckForMissing;
     private final long maxSpanInNanos;
 
     private final boolean descending;
 
     private final Limit limit;
+    private final boolean[] missingEvents;
+    protected final int firstPositiveStage;
+    protected final int lastPositiveStage;
+    private final boolean missingEventsExist;
```

Comment:
`missingEventQueriesExist`
`missingEventStagesExist`
`missingEventFiltersExist`
`missingEventCriteriaExist` (less favorable on this one because the "criteria" is used in other classes, this one is more about "stages")

Reply:
I like `missingEventStagesExist`